Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -288,44 +468,49 @@ void CDMInstanceOpenCDM::requestLicense(LicenseType licenseType, const AtomicStr
 
     GST_TRACE("Going to request a new session id, init data size %u and MD5 %s", initData.sizeInBytes(), GStreamerEMEUtilities::initDataMD5(initData).utf8().data());
     GST_MEMDUMP("init data", initData.characters8(), initData.sizeInBytes());
+    auto generateChallenge = [this, callback = WTFMove(callback)](Session* session) {
+        auto sessionId = session->id();
+        if (sessionId.isEmpty()) {
+            GST_ERROR("could not create session id");
+            callback(session->initData()->copy(), { }, false, Failed);
```

Comment:
Now that I see the issues with moving the buffers when calling the callbacks, I realized we could do `session->initData().copyRef()` instead of copying the whole shared buffer. Search&replace, plz :sweat_smile: